pennylane/tape/circuit_graph.py

```diff
@@ -30,6 +30,7 @@ class TapeCircuitGraph(CircuitGraph):
     def __init__(self, ops, obs, wires):
         self._operations = ops
         self._observables = obs
+        self._depth = 0
```

Comment:
I steer towards using `None` for attributes that are not yet set/computed (can be ambigious whether `self._depth == 0` indicates that it hasn't been computed yet, or whether it has been computed (and found to be zero). Also leads to problems if developers use the pattern `if self._depth: ...`

Reply:
Oh I see, that makes sense! :+1: